Templates/integration_template.ps1

```diff
@@ -0,0 +1,83 @@
+<#
+.Synopsis
+   Template for creating DSC Resource Integration Tests
+.DESCRIPTION
+   To Use:
+     1. Copy to \Tests\Integration\ folder and rename MSFT_x<ResourceName>.Integration.tests.ps1
+     2. Customize TODO sections.
+     3. Create test DSC Configurtion file MSFT_x<ResourceName>.config.ps1 from integration_config_template.ps1 file.
+
+.NOTES
+   Code in HEADER, FOOTER and DEFAULT TEST regions are standard and may be moved into
+   DSCResource.Tools in Future and therefore should not be altered if possible.
+#>
+
+# TODO: Customize these paramters...
+$DSCModuleName      = 'x<ModuleName>' # Example xNetworking
+$DSCResourceName    = 'MSFT_x<ResourceName>' # Example MSFT_xFirewall
+# /TODO
+
+#region HEADER
+if ( (-not (Test-Path -Path '.\DSCResource.Tests\')) -or `
+     (-not (Test-Path -Path '.\DSCResource.Tests\TestHelper.psm1')) )
+{
+    Throw @(
+        "The DSCResource.Tests folder could not be found in the root folder of the $DSCModuleName DSC Module to test."
+        "Please use Git to clone this repository to the root folder of the $DSCModuleName DSC Module that needs to be tested using the command:"
+        "git clone https://github.com/PowerShell/DscResource.Tests.git"
+    ) -join "`n"
+}
+Import-Module .\DSCResource.Tests\TestHelper.psm1 -Force
+$TestEnvironment = Initialize-TestEnvironment `
+    -DSCModuleName $DSCModuleName `
+    -DSCResourceName $DSCResourceName `
+    -TestType Integration 
+#endregion
+
+# TODO: Other Init Code Goes Here...
+
+# Using try/finally to always cleanup even if something awful happens.
+try
+{
+
+
+    #region Integration Tests
+    <#
+      This file exists so we can load the test file without necessarily having xNetworking in
+      the $env:PSModulePath. Otherwise PowerShell will throw an error when reading the Pester File.
+    #>
+    $ConfigFile = Join-Path -Path $PSScriptRoot -ChildPath "$DSCResourceName.config.ps1"
+    . $ConfigFile
+
+    Describe "$($DSCResourceName)_Integration" {
+        #region DEFAULT TESTS
+        It 'Should compile without throwing' {
+            {
+                [System.Environment]::SetEnvironmentVariable('PSModulePath',
+                    $env:PSModulePath,[System.EnvironmentVariableTarget]::Machine)
+                Invoke-Expression -Command "$($DSCResourceName)_Config -OutputPath `$TestEnvironment.WorkingFolder"
```

Comment:
Good catch! I've changed the config name in the config template to match this rather than change the code here so that it is more obvious what is being called.